pkg/containermanager/docker/docker_manager.go

```diff
@@ -30,7 +30,7 @@ func NewDockerContainerManager(logger lager.Logger, client DockerClient) contain
 	}
 }
 
-func (dc dockerContainerManager) Provision(ctx context.Context, containerConfig *containermanager.ContainerConfig) error {
+func (dc dockerContainerManager) Provision(ctx context.Context, containerConfig containermanager.ContainerConfig) error {
```

Comment:
is there a reason for 'not a pointer' we can pull out into a separate pr to reduce review churn?

Reply:
yeah I made that part of a separate commit to maybe later on split into a separate PR. generally I think configs should be treated as immutable objects.

I had it as a meeting topic for us to discuss since I have noticed that I usually make objects not be pointers unless needed, and that they get changed somehow during the process. Would be good to establish a team convention to avoid drift.

Reply:
I don't think there's a good go way to do that exact thing. The closet thing I can think of is to make a config interface with accessors.

pointer has the concept of you can 'accidentally' set something in the passed in struct, and copy makes a copy and can be confusing about why you changed something and it did not change.